test/object-store/sync/app.cpp

```diff
@@ -2708,16 +2708,11 @@ TEST_CASE("app: sync integration", "[sync][app]") {
     SECTION("too large sync message error handling") {
         SyncTestFile config(app, partition, schema);
 
-        std::mutex sync_error_mutex;
-        bool done = false;
-        config.sync_config->error_handler = [&](auto, SyncError error) {
-            if (error.error_code.category() != util::websocket::websocket_close_status_category())
-                return;
-            std::lock_guard<std::mutex> lk(sync_error_mutex);
-            done = true;
-            REQUIRE(error.error_code.value() == 1009);
-            REQUIRE(error.message == "read limited at 16777217 bytes");
-        };
+        auto pf = util::make_promise_future<SyncError>();
+        config.sync_config->error_handler =
+            [sp = util::CopyablePromiseHolder(std::move(pf.promise))](auto, SyncError error) mutable {
```

Comment:
Should we always use this when passing a promise into the lambda captures? Does `sp = std::move(pf.promise)` not work like expected?

Reply:
it depends on what the lambda is going to be used for. If you need to store the lambda in a std::function, where all captures are required to be copyable, then you need to wrap a Promise (which is not copyable) into something that is. If you're storing the lambda in a util::UniqueFunction (which requires all its captures to be moveable and not copyable) then you don't need this. Likewise, if you aren't storing your lambda in any type-erasing wrapper at all, you don't need this.

Reply:
Thank you for the explanation!